sdks/python/apache_beam/ml/inference/sklearn_loader.py

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+import enum
+import pickle
+import sys
+from dataclasses import dataclass
+from typing import Any
+from typing import Iterable
+from typing import List
+
+import joblib
+import numpy
+
+import apache_beam.ml.inference.api as api
+import apache_beam.ml.inference.base as base
+import sklearn_loader
+from apache_beam.io.filesystems import FileSystems
+
+
+class SerializationType(enum.Enum):
+  PICKLE = 1
+  JOBLIB = 2
+
+
+class SKLearnInferenceRunner(base.InferenceRunner):
+  def run_inference(self, batch: List[numpy.array],
+                    model: Any) -> Iterable[numpy.array]:
+    # vectorize data for better performance
+    vectorized_batch = numpy.stack(batch, axis=0)
+    predictions = model.predict(vectorized_batch)
+    return [api.PredictionResult(x, y) for x, y in zip(batch, predictions)]
+
+  def get_num_bytes(self, batch: List[numpy.array]) -> int:
+    """Returns the number of bytes of data for a batch."""
+    return sum(sys.getsizeof(element) for element in batch)
+
+
+class SKLearnModelLoader(base.ModelLoader):
+  def __init__(
+      self,
+      serialization: SerializationType = SerializationType.PICKLE,
+      model_uri: str = ''):
+    self._serialization = serialization
+    self._model_uri = model_uri
```

Comment:
Should we try to be as consistent as possible across frameworks? For example, for Pytorch, I use `state_dict_path`, but I could change it to be `state_dict_uri`.

Reply:
I think it's fine to use different parameters. In my view, path communicates more a directory, and URI a single file.

TFX-BSL is using saved_model_spec (which is a proto) that contains model_path. As far as I can tell, TF saves models to a path rather than a URI or file.
https://cloud.google.com/blog/topics/developers-practitioners/using-tfx-inference-dataflow-large-scale-ml-inference-patterns

What does pytorch do? Is it a single file or a path with a bunch of data?

Reply:
The standard Pytorch way is just a single file storing the model spec parameters.

I'll probably stick with `state_dict_path` because that's the Pytorch [convention](https://pytorch.org/tutorials/beginner/saving_loading_models.html) of storing saved models.

Reply:
It sounds like a single filepath is the appropriate interface for both sklearn and pytorch, why not be consistent?

Reply:
I agree.

A state dict is a python dictionary object used by pytorch. It's strange to use that name for sklearn. I like model uri, model_path is fine too.